Source/Core/Core/AchievementManager.cpp

```diff
@@ -171,12 +173,31 @@ void AchievementManager::ActivateDeactivateAchievements()
   }
 }
 
+void AchievementManager::ActivateDeactivateLeaderboards()
+{
+  bool leaderboards_enabled = Config::Get(Config::RA_LEADERBOARDS_ENABLED);
+  for (u32 ix = 0; ix < m_game_data.num_leaderboards; ix++)
+  {
+    auto leaderboard = m_game_data.leaderboards[ix];
+    if (m_is_game_loaded && leaderboards_enabled && hardcore_mode_enabled)
+    {
+      if (rc_runtime_get_lboard(&m_runtime, leaderboard.id) == nullptr)
```

Comment:
Are you sure you want this function? [The wiki entry for it](https://github.com/RetroAchievements/rcheevos/wiki/rc_runtime_get_lboard) says it should primarily be used for debugging.

Reply:
Investigating. Problem is that the wiki also says activating an already-active achievement or lboard will reset it, so I need to verify that it's active, and there's no other function that does that; I'd have to iterate through the data structure, which I don't have to do anywhere else.